python/ray/dag/compiled_dag_node.py

```diff
@@ -806,6 +808,15 @@ def _preprocess(self) -> None:
                 if dag_node.type_hint.requires_nccl():
                     # Add all writers to the NCCL group.
                     nccl_actors.add(actor_handle)
+                    custom_nccl_group = dag_node.type_hint.get_custom_nccl_group()
+                    if custom_nccl_group is not None:
+                        if self._custom_nccl_group is not None:
+                            assert self._custom_nccl_group == custom_nccl_group, (
+                                "Accelerated DAGs currently only support "
+                                "a single custom NCCL group, but multiple "
+                                "have been specified."
```

Comment:
can you print the nccl group here?

Reply:
We don't currently enforce having a meaningful `__str__` implementation for nccl groups, so it may not help identifying the issues (say when address of the object is printed).

Instead I added the following to make the error message more actionable:
"Check all the TorchTensor(transport=nccl_group) type hints to make sure only one NCCL group is used."